src/js/actions/history.js

```diff
@@ -279,6 +283,7 @@ define(function (require, exports) {
     };
     onReset.reads = [];
     onReset.writes = [];
+    onReset.modal = true;
```

Comment:
Basically, it seems like all the onReset actions should be consistent in that regard. It also seems like maybe the `FluxController` should just kill modal states before we call `onReset`.